packages/shared/src/hooks/referral/useReferralReminder.ts

```diff
@@ -0,0 +1,31 @@
+import { useContext, useEffect } from 'react';
+import { useLazyModal } from '../useLazyModal';
+import { LazyModal } from '../../components/modals/common/types';
+import AlertContext from '../../contexts/AlertContext';
+import useDebounce from '../useDebounce';
+
+export const useReferralReminder = (): void => {
+  const { alerts, updateLastReferralReminder } = useContext(AlertContext);
+  const { openModal } = useLazyModal();
+  // We need a small debounce to avoid it not being opened due to fast re-renders
```

Comment:
When does this happen exactly? Fixing race conditions with delays usually fails just later. I'd be happy to take a look if needed.

Reply:
Just on main feed :O 
Remove debounce and simply open modal and it happens.

I got it working with other evaluation criteria, but felt even worse.
Would love to hear other thoughts

Reply:
This is very interesting, I wonder what could be the root cause of this issue. Could it be the main content getting mounted twice, hence the lazy modal element is getting removed from the initial instance?

Reply:
We tried to debug it for a while, no idea.
Disabling the query (which is safe for this one) works weird enough.

Reply:
It's a combination of react 17 and react-query.. sandbox worked because it was react 18, here is a sandbox that does not work https://codesandbox.io/s/admiring-surf-mjjtq4?file=/src/App.js

On 4.x and beyond it works https://codesandbox.io/s/zealous-saha-s9dzxm?file=/src/App.js

On 5.x it works also but looks like 5.x needs react 18 because it uses useSyncExternalStore hook